src/shaders/fill_extrusion.fragment.glsl

```diff
@@ -1,7 +1,24 @@
 varying vec4 v_color;
 
+#ifdef FAUX_AO
+uniform lowp vec2 u_ao;
+varying vec3 v_ao;
+#endif
+
 void main() {
     vec4 color = v_color;
+#ifdef FAUX_AO
+    float intensity = u_ao[0];
+    float h = max(0.0, v_ao.z);
+    float h_floors = h / u_ao[1];
+    float y_shade = 1.0 - 0.9 * intensity * min(v_ao.y, 1.0);
+    float shade = (1.0 - 0.08 * intensity) * (y_shade + (1.0 - y_shade) * (1.0 - pow(1.0 - min(h_floors / 16.0, 1.0), 16.0))) + 0.08 * intensity * min(h_floors / 160.0, 1.0);
+    // concave angle
+    float concave = v_ao.x * v_ao.x;
+    float x_shade = mix(1.0, mix(0.6, 0.75, min(h_floors / 30.0, 1.0)), intensity) + 0.1 * intensity * min(h, 1.0);
+    shade *= mix(1.0, x_shade * x_shade * x_shade, concave);
+    color.rgb = color.rgb * shade;
+#endif
```

Comment:
Could you add some descriptions to magic numbers used here?

Reply:
https://github.com/mapbox/mapbox-gl-js/pull/11926#discussion_r898496259

Reply:
I do see the explanation in the comment message but there's no reference to the documentation in the code. I was thinking about making this more easily approachable in the future as the location of the documentation is not immediately clear